modules/tensor_mechanics/src/materials/ComputeLagrangianStressStVenantKirchhoff.C

```diff
@@ -0,0 +1,94 @@
+#include "ComputeLagrangianStressStVenantKirchhoff.h"
+
+registerMooseObject("TensorMechanicsApp", ComputeLagrangianStressStVenantKirchhoff);
+
+InputParameters
+ComputeLagrangianStressStVenantKirchhoff::validParams()
+{
+  InputParameters params = ComputeLagrangianStressPK::validParams();
+
+  params.addParam<Real>("mu", "Lame parameter mu");
+  params.addParam<Real>("lambda", "Lame parameter lambda");
+
+  return params;
+}
+
+ComputeLagrangianStressStVenantKirchhoff::ComputeLagrangianStressStVenantKirchhoff(
+    const InputParameters & parameters)
+  : ComputeLagrangianStressPK(parameters),
+    _mu(getParam<Real>("mu")),
+    _lambda(getParam<Real>("lambda"))
+{
+}
+
+void
+ComputeLagrangianStressStVenantKirchhoff::computeQpPKStress()
+{
+  // Hyperelasticity is weird, we need to branch on the type of update if we
+  // want a truly linear model
+  //
+  // Model with nonlinear strain
+  if (_ld)
```

Comment:
Seeing this makes me think we should probably use the `Guarantee` system to define the "large deformation kinematics" parameter. That way a stress calculator only needs to concern one of the `_ld = true` or `_ld = false` case, not both. @dschwen what do you think?

Reply:
For example, in this case, I think it makes sense for this SVK model to only deal with `_ld = true`, because if `_ld = false` it reduces to `ComputeLagrangianElasticSmallStress`.

Reply:
For some models (notably standard, small strain models) it makes sense to allow `_ld = true` or `_ld = false`.  You code the small deformation update, but the class knows to use an objective update if `_ld = true` or just the base formulation if `_ld = false`.  But maybe I don't understand the constraint system, I haven't used it.